heron/proto/ckptmgr.proto

```diff
@@ -0,0 +1,200 @@
+// start processing.
+message StartStmgrStatefulProcessing {
+  required string checkpoint_id = 1;
+}
+
+// tmaster -> ckptmgr messages
+
+// This is the message that a tmaster sends
+// when it wants to establish the connection
+// with checkpoint manager
+message RegisterTMasterRequest{
+  required string topology_name = 1;
+  required string topology_id = 2;
+}
+
+// This is the message that checkpoint manager
+// sends when it receives the register request
+// from tmaster
+message RegisterTMasterResponse {
+  required heron.proto.system.Status status = 1;
+}
+
+// Message sent by tmaster to ckptmgr to cleanup
+// old checkpoint state.
+message CleanStatefulCheckpointRequest {
+  // Any checkpoints older than this can be cleaned
+  optional string oldest_checkpoint_preserved = 1;
+  optional bool clean_all_checkpoints = 2 [default = false];
+}
+
+// Message sent by ckptmgr to tmaster about the cleanup request
+message CleanStatefulCheckpointResponse {
+  required heron.proto.system.Status status = 1;
+  repeated string cleaned_checkpoints = 2;
+}
+
+// stmgr -> ckptmgr messages
+
+// This is the message that a stmgr sends
+// when it wants to establish the connection
+// with checkpoint manager
+message RegisterStMgrRequest{
+  required string topology_name = 1;
+  required string topology_id = 2;
+  required string stmgr_id = 3;
+}
+
+// This is the message that checkpoint manager
+// sends when it receives the register request
+// from stmgr
+message RegisterStMgrResponse {
+  required heron.proto.system.Status status = 1;
+}
+
+// This is the request that StMgr sends to Checkpoint Mgr
+// to store instance checkpoint data
+message SaveInstanceStateRequest {
+  // Information about the instance whose state this is
+  required heron.proto.system.Instance instance = 1;
+  required InstanceStateCheckpoint checkpoint = 2;
+}
+
+// After successfully saving state, this is what
+// Checkpoint Mgr sends back as response
+message SaveInstanceStateResponse {
+  required heron.proto.system.Status status = 1;
+  required heron.proto.system.Instance instance = 2;
+  required string checkpoint_id = 3;
+}
+
+// This is the request that StMgr sends to Checkpoint Mgr
+// to fetch the instance state data
+message GetInstanceStateRequest {
+  required heron.proto.system.Instance instance = 1;
+  required string checkpoint_id = 2;
+}
+
+// This is the response that Checkpoint Mgr sends to StMgr
+// for its get instance state request
+message GetInstanceStateResponse {
+  required heron.proto.system.Status status = 1;
+  required heron.proto.system.Instance instance = 2;
+  required string checkpoint_id = 3;
+  optional InstanceStateCheckpoint checkpoint = 4;
+}
+
+// stmgr -> Instance messages
+
+// This is the message that the stmgr sends to its
+// local tasks to begin initiating checkpoint
+message InitiateStatefulCheckpoint {
+  required string checkpoint_id = 1;
+}
+
+// This is the message that the instance sends
```

Comment:
Is this the message or is this the state info that's part of another message? Reading above it seems the latter but the description states the former. If it is the former, can we put a verb in this object, to make it more clear what action is being requested?